### PR TITLE
Disabling bindings by default in cmake.

### PR DESCRIPTION
**Pull request by RishabhGarg108**
This PR tries to solve #2773 .
The default options for all bindings are set to "OFF" in this.

### CODE REVIEW

**Review by Yashwants19 — APPROVED** [review]
This looks good to me. Can you also update `HISTORY.md`?

**Comment by RishabhGarg108** [comment]
@Yashwants19 , I have updated `HISTORY.md`.
Thanks for review :)

**Review by shrit — APPROVED** [review]
Looks good to me:+1:

**Comment by RishabhGarg108** [comment]
Thanks for review @shrit :-D

**Comment by RishabhGarg108** [comment]
I am wondering why one test is now failing. Previously it was not failing.

**Comment by zoq** [comment]
> I am wondering why one test is now failing. Previously it was not failing.

Some tests are non deterministic, so it's possible that they can fail some time e.g. if the threshold is to tight.

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### mlpack ?.?.?
 ###### ????-??-??
+  * Disabled all the bindings by default in cmake (#2782).
```
Comment (zoq) [inline HISTORY.md:3]:
```suggestion
  * Disabled all the bindings by default in cmake (#2782).

```

Minor style fix, add an empty line between the updates.

**Comment by RishabhGarg108** [comment]
> > I am wondering why one test is now failing. Previously it was not failing.
> 
> Some tests are non deterministic, so it's possible that they can fail some time e.g. if the threshold is to tight.

Okay :+1:  

I have fixed the style. Thanks for review :)

**Comment by Yashwants19** [comment]
Thanks @RishabhGarg108 for looking into the issue. :+1:

**Pull request merged**